### DISCUSSION TITLE
How can I give the FileOpenPicker a flag for android? (GrantPersistableUriPermission)

### DISCUSSION THREAD

**Discussion by jp-weber** [unoplatform/uno#10536] [Q&A]
I want add for the Android FileOpenPicker the intent flag for the persistent access. Is there any way to do this without rebuilding the function? The current problem is that with the fileopenpicker, persistent access to the files is not possible, which is required for my app.
https://github.com/unoplatform/uno/blob/56d3ba98ee6cab055dc8d7c4941ce4494e20e946/src/Uno.UWP/Storage/Pickers/FileOpenPicker.Android.cs#L59

Set
```csharp
intent.AddFlags(Android.Content.ActivityFlags.GrantPersistableUriPermission);
```
and for the result =>
```csharp
Android.App.Application.Context.ContentResolver.TakePersistableUriPermission(androidUri, Android.Content.ActivityFlags.GrantPersistableUriPermission);
```

**Comment by jeromelaban** [comment]
There isn't any at this time, be it could be possible provide a way to access the intent before it is started.

Something like:
```csharp
var disposable = FileOpenPickerHelper.RegisterOnBeforeStartActivity(picker, intent => intent.AddFlags(Android.Content.ActivityFlags.GrantPersistableUriPermission));
```

Which would set allow the user to set a callback directly into `FileOpenPicker` for android.

**Reply by jp-weber** [reply]
Oh that would be awesome 😊

**Reply by jeromelaban** [reply]
Let's track in https://github.com/unoplatform/uno/issues/10550
